cms/plugin_base.py

```diff
@@ -431,19 +441,55 @@ def get_fieldsets(self, request, obj=None):
 
         return fieldsets
 
-    def get_child_classes(self, slot, page):
+    @classmethod
+    def get_child_class_overrides(cls, slot, page):
+        """
+        Returns a list of plugin types that are allowed
+        as children of this plugin.
+        """
         from cms.utils.placeholder import get_placeholder_conf
 
         template = page and page.get_template() or None
 
         # config overrides..
         ph_conf = get_placeholder_conf('child_classes', slot, template, default={})
-        child_classes = ph_conf.get(self.__class__.__name__, self.child_classes)
+        return ph_conf.get(cls.__name__, cls.child_classes)
+
+    @classmethod
+    def get_child_plugin_candidates(cls, slot, page):
+        """
+        Returns a list of all plugin classes
+        that will be considered when fetching
+        all available child classes for this plugin.
+        """
+        # By adding this as a separate method,
+        # we allow other plugins to affect the list of child plugin candidates.
+        # Useful in cases like djangocms-text-ckeditor where only text only
+        # plugins are allowed.
```

Comment:
Very minor, but even though we have 120 chars for Python, the comments are most readable if they're formatted to fit 80 columns. In this case, it looks like you chopped out some other text, but could you re-justify it into an 80-col block?